### PR TITLE
DOMA-586 Make error handling more obvious

### PR DESCRIPTION
**Pull request by toplenboren**
## Problem

Sometimes we get strange programmatic errors from `registerServiceConsumer` mutation
- `cannot read id of null` 
- `cannot read id of undefined`

## Solution

Refactor the code a bit

### CODE REVIEW

File: apps/condo/domains/resident/schema/RegisterServiceConsumerService.js [hunk 1]
```diff
@@ -65,10 +65,15 @@ const RegisterServiceConsumerService = new GQLCustomSchema('RegisterServiceConsu
...
 
-                if (!Array.isArray(applicableBillingAccounts)) {
-                    throw new Error(`${NOT_FOUND_ERROR}billingAccount] No suitable billing accounts found for this user`)
+                // todo(toplenboren) remove this once B2C integration case is ready
+                if (resident.organization) {
+                    applicableBillingAccounts = await getAccountsFromOrganizationIntegration(context, resident, unitName, accountNumber)
+                }
+
+                if (!Array.isArray(applicableBillingAccounts) || applicableBillingAccounts.length === 0) {
+                    throw new Error(`${NOT_FOUND_ERROR}account] BillingAccounts not found for this user`)
```
Comment (SavelevMatthew) [inline apps/condo/domains/resident/schema/RegisterServiceConsumerService.js:76]: Why throwing error here? Why not returning [] when resident does not have BillingAccounts?

File: apps/condo/domains/resident/schema/RegisterServiceConsumerService.test.js [hunk 2]
```diff
@@ -38,6 +38,35 @@ describe('RegisterServiceConsumerService', () => {
...
+        const payload = {
+            residentId: resident.id,
+            unitName: billingAccountAttrs.unitName,
+            accountNumber: billingAccountAttrs.number,
+        }
+
+        await catchErrorFrom(async () => {
+            await registerServiceConsumerByTestClient(userClient, payload)
+        }, (e) => {
+            expect(e.errors[0].message).toContain('BillingAccounts not found')
```
Comment (SavelevMatthew) [inline apps/condo/domains/resident/schema/RegisterServiceConsumerService.test.js:66]: Empty list?

File: apps/condo/domains/resident/schema/RegisterServiceConsumerService.js (hunk 1 above)
Comment (toplenboren) [reply apps/condo/domains/resident/schema/RegisterServiceConsumerService.js:76]: We need to somehow show error that serviceConsumer cannot be created.

It happens on the mobile screen when user is prompted with

"Введите свой лицевой счет"

If Account is not valid, I think its better signalize, not return empty arr

File: apps/condo/domains/resident/schema/RegisterServiceConsumerService.test.js (hunk 2 above)
Comment (toplenboren) [reply apps/condo/domains/resident/schema/RegisterServiceConsumerService.test.js:66]: Answered in previous comment

**Review by SavelevMatthew — APPROVED** [review]

**Review by dkoviazin — APPROVED** [review]

**Pull request merged**